anta/tests/routing/bgp.py

```diff
@@ -447,3 +448,213 @@ def test(self) -> None:
 
         if failures:
             self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPPeerMPCaps(AntaTest):
+    """
+    Verifies the multiprotocol capabilities of a BGP peer in a specified VRF.
+    Expected results:
+        * success: The test will pass if the BGP peer's multiprotocol capabilities are advertised, received, and enabled in the specified VRF.
+        * failure: The test will fail if BGP peers are not found or multiprotocol capabilities are not advertised, received, and enabled in the specified VRF.
+    """
+
+    name = "VerifyBGPPeerMPCaps"
+    description = "Verifies the multiprotocol capabilities of a BGP peer in a specified VRF"
+    categories = ["routing", "bgp"]
+    commands = [AntaCommand(command="show bgp neighbors")]
+
+    class Input(AntaTest.Input):
+        """
+        Input parameters of the testcase.
+        """
+
+        bgp_peers: List[BgpPeers]
+        """List of BGP peers"""
+
+        class BgpPeers(BaseModel):
+            """
+            This class defines the details of a BGP peer.
+            """
+
+            peer: Union[IPv4Address, IPv6Address]
+            """IPv4/IPv6 BGP peer"""
+            vrf: str = "default"
+            """VRF context"""
+            capabilities: List[MultiProtocolCaps]
+            """Multiprotocol Capabilities"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failures: dict[str, Any] = {}
+
+        # Iterate over each bgp peer
+        for bgp_peer in self.inputs.bgp_peers:
+            peer = str(bgp_peer.peer)
+            vrf = bgp_peer.vrf
+            capabilities = bgp_peer.capabilities
+
+            # Check if BGP output exists
+            if not (bgp_output := get_value(self.instance_commands[0].json_output, f"vrfs.{vrf}.peerList")):
+                failures.setdefault("bgp_peers", {})[str(peer)] = {vrf: "Not configured"}
+                continue
+
+            bgp_index = create_index(bgp_output, "peerAddress")
+            bgp_output = bgp_index.get(peer)
+            if not bgp_output:
+                failures.setdefault("bgp_peers", {})[str(peer)] = {vrf: "Not configured"}
```

Comment:
if a input peer is not found in list of dict then it will raise a key error so to handle this need to check again. First one will check if peers are configured with required VRF